crates/shared/src/price_estimation/native_price_cache.rs

```diff
@@ -341,6 +381,39 @@ mod tests {
         H160::from_low_u64_be(u)
     }
 
+    #[test]
+    fn computes_price_from_normalized_price() {
+        assert_eq!(
+            CachingNativePriceEstimator::from_normalized_price(U256::from(
+                500_000_000_000_000_000_u128
+            ))
+            .unwrap(),
+            0.5
+        );
+    }
+
+    #[tokio::test]
+    async fn caches_successful_estimates_with_loaded_prices() {
+        let mut inner = MockNativePriceEstimating::new();
+        inner.expect_estimate_native_price().never();
+
+        let prices = BTreeMap::from([(token(0), U256::from(1_000_000_000_000_000_000_u128))]);
+        let estimator = CachingNativePriceEstimator::new(
+            Box::new(inner),
+            Duration::from_millis(30),
+            Default::default(),
+            None,
+            Default::default(),
+            1,
+            Some(prices),
+        );
+
+        for _ in 0..10 {
+            let result = estimator.estimate_native_price(token(0)).await;
+            assert!(result.as_ref().unwrap().to_i64().unwrap() == 1);
```

Comment:
To get slightly nicer error messages when the assertion fails.
```suggestion
            assert_eq!(result.as_ref().unwrap().to_i64().unwrap(), 1);
```